### PR TITLE
mcux: flexcom: i2c: Fix slave report

### PR DESCRIPTION
**Pull request by dleach02**
Patch to address driver not properly reporting if slave node is present on bus.

Fixes #57858

### CODE REVIEW

**Comment by danieldegrasse** [comment]
Assuming this is the bug this PR refers to: https://github.com/zephyrproject-rtos/zephyr/issues/57858

**Comment by danieldegrasse** [comment]
Can we place a note in the README file describing this patch and clarifying the bug should be fixed by the next SDK update?

**Review by hakehuang — APPROVED** [review]
Test PASS and commnets leave on 
https://github.com/zephyrproject-rtos/zephyr/issues/57858

File: mcux/mcux-sdk/drivers/flexcomm/fsl_i2c.c [hunk 1]
```diff
@@ -887,7 +897,7 @@ status_t I2C_MasterTransferNonBlocking(I2C_Type *base, i2c_master_handle_t *hand
     result = I2C_InitTransferStateMachine(base, handle, xfer);
 
     /* Clear error flags. */
-    I2C_ClearStatusFlags(base, I2C_STAT_MSTARBLOSS_MASK | I2C_STAT_MSTSTSTPERR_MASK);
+    I2C_ClearStatusFlags(base, (uint32_t)((uint32_t)kI2C_MasterAllClearFlags | (uint32_t)kI2C_CommonAllClearFlags));
```
Comment (hakehuang) [inline mcux/mcux-sdk/drivers/flexcomm/fsl_i2c.c:900]:
```suggestion
    I2C_ClearStatusFlags(base, (uint32_t)(kI2C_MasterAllClearFlags | kI2C_CommonAllClearFlags));
```

Comment (dleach02) [reply mcux/mcux-sdk/drivers/flexcomm/fsl_i2c.c:900]: @hakehuang, this is coming directly from the internal SDK. Can you take this up internally with the SDK team?

File: mcux/README [hunk 2]
```diff
@@ -60,12 +60,14 @@ License Link:
...
      be four being used for entropy in zephyr.
-  3. Fixed flexcan_fd_frame being written into/written from the message buffer in the flexcan driver (mcux/mcux-sdk/drivers/flexcan/fsl_flexcan.c). On the
+  3. mcux-sdk/drivers/flexcan/fsl_flexcan.c: Fixed flexcan_fd_frame being written into/written from the message buffer in the flexcan driver. On the
      write side, the EDL bit is now utilized for selection of can frame type and on the receiver side, the EDL and BRS status are read from the message
      buffer.
   4. Add device_system cmake definitions for the following SOCs: MKL25Z4, MK82F25615, MKW24D5, MKW40Z4, MKW41Z4
   5. Fixed the FlexCAN driver to propagate kStatus_FLEXCAN_RxOverflow mailbox status when using the FlexCAN driver transactional APIs.
   6. Fixed fsl_caam.c: CAAM_RNG_GetRandomDataNonBlocking() to not force a reseed with each request. Internal bug submitted [MCUX-57074]
   7. fsl_common.h: add #ifdef ZEPHYR #endif to include Zephyr's sys utils
+  8. mcux-sdk/drivers/lpuart/fsl_lpuart.h: modified return type of LPUART_GetDataRegisterAddress() to be uintptr_t for 64 bit architectures. Internal change submitted.
```
Comment (mmahadevan108) [inline mcux/README:72]: I don't see this change in this PR, is this needed?

Comment (dleach02) [reply mcux/README:72]: I think what happened was that this readme was supposed to be included in the last HAL update I did. I must have missed including it.

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**